admin_panel/students/viewsets/student.py

```diff
@@ -0,0 +1,90 @@
+from rest_framework import viewsets, filters
+from django_filters.rest_framework import DjangoFilterBackend
+from drf_yasg.utils import swagger_auto_schema
+from drf_yasg import openapi
+
+from admin_panel.models import Student
+from admin_panel.students.serializers.student import StudentResponseSerializer, StudentSerializer
+
+
+class StudentViewset(viewsets.ModelViewSet):
+    """
+    API endpoints for managing students in the system.
+    Supports searching by first_name and last_name, and filtering on all fields.
+    """
+    queryset = Student.objects.all()
+    serializer_class = StudentSerializer
+    filter_backends = [filters.SearchFilter, DjangoFilterBackend]
+    search_fields = ['first_name', 'last_name']
+    filterset_fields = '__all__'
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider restricting filterset fields</b></div>
<div id="fix">

Consider restricting `filterset_fields = '__all__'` to specific fields that are needed for filtering. Using `'__all__'` could expose sensitive data and impact performance.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -19,1 +19,1 @@
 -    filterset_fields = '__all__'
 +    filterset_fields = ['first_name', 'last_name', 'email', 'grade', 'section', 'batch', 'active']
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/isshiki-examevokee/ExamEvoke/pull/2#issuecomment-2496536252>#ae3661</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged